src/app/DIAmeterApplication.cpp

```diff
@@ -0,0 +1,558 @@
+
+  const pb::Header::PeptidesHeader& pepHeader = peptides_header.peptides_header();
+  DECOY_TYPE_T headerDecoyType = (DECOY_TYPE_T)pepHeader.decoys();
+  int decoysPerTarget = pepHeader.has_decoys_per_target() ? pepHeader.decoys_per_target() : 0;
+  carp(CARP_DEBUG, "decoysPerTarget = %d \t headerDecoyType=%d.", decoysPerTarget, headerDecoyType);
+
+  MassConstants::Init(&pepHeader.mods(), &pepHeader.nterm_mods(), &pepHeader.cterm_mods(), &pepHeader.nprotterm_mods(), &pepHeader.cprotterm_mods(), bin_width_, bin_offset_);
+  ModificationDefinition::ClearAll();
+  TideMatchSet::initModMap(pepHeader.mods(), ANY);
+  TideMatchSet::initModMap(pepHeader.nterm_mods(), PEPTIDE_N);
+  TideMatchSet::initModMap(pepHeader.cterm_mods(), PEPTIDE_C);
+  TideMatchSet::initModMap(pepHeader.nprotterm_mods(), PROTEIN_N);
+  TideMatchSet::initModMap(pepHeader.cprotterm_mods(), PROTEIN_C);
+
+
+  string output_file_name_ = make_file_path("tide-search.txt");
+  ofstream* output_file = create_stream_in_path(output_file_name_.c_str(), NULL, Params::GetBool("overwrite"));
+  TideMatchSet::writeHeadersDIA(output_file, Params::GetBool("compute-sp"));
+
+  map<string, double> peptide_predrt_map;
+  getPeptidePredRTMapping(&peptide_predrt_map);
+
+  /*
+  vector<InputFile> ms1_spectra_files = getInputFiles(input_files, 1);
+  vector<InputFile> ms2_spectra_files = getInputFiles(input_files, 2);
+
+
+  // Loop through spectrum files
+  for (pair<vector<InputFile>::const_iterator, vector<InputFile>::const_iterator> f(ms1_spectra_files.begin(), ms2_spectra_files.begin());
+		  f.first != ms1_spectra_files.end() && f.second != ms2_spectra_files.end(); ++f.first, ++f.second) {
+
+	  string ms1_spectra_file = (f.first)->SpectrumRecords;
+	  string ms2_spectra_file = (f.second)->SpectrumRecords;
+	  string origin_file = (f.second)->OriginalName;
+
+	  map<int, pair<double*, double*>> ms1scan_intensity_rank_map;
+	  avg_noise_intensity_logrank_ = loadMS1Spectra(ms1_spectra_file, &ms1scan_intensity_rank_map);
+	  SpectrumCollection* spectra = loadSpectra(ms2_spectra_file);
+
+	  // insert the search code here and will split into a new function later
+	  double highest_ms2_mz = spectra->FindHighestMZ();
+	  carp(CARP_DEBUG, "Maximum observed MS2 m/z = %f.", highest_ms2_mz);
+	  MaxBin::SetGlobalMax(highest_ms2_mz);
+
+
+	  // Active queue to process the indexed peptides
+	  ActivePeptideQueue* active_peptide_queue = new ActivePeptideQueue(peptide_reader->Reader(), proteins);
+	  active_peptide_queue->SetBinSize(bin_width_, bin_offset_);
+	  active_peptide_queue->SetOutputs(NULL, &locations, Params::GetInt("top-match"), true, output_file, NULL, highest_ms2_mz);
+
+	  // initialize fields required for output
+	  // Not sure if it is necessary, will check later
+	  const vector<SpectrumCollection::SpecCharge>* spec_charges = spectra->SpecCharges();
+	  int* sc_index = new int(-1);
```

Comment:
does this need to be a pointer?

Reply:
Not necessary, but just follow the same as TideSearchApplication